### DISCUSSION TITLE
Opencast recording ingestion support

### DISCUSSION THREAD

**Discussion by Lyx52** [mynaparrot/plugNmeet-server#369] [Ideas]
Hi I had idea for a new feature, "automatic recording uploads into opencast video management system". There are few ways this can be implemented

- Simplest way is if plugNmeet-recorder recording ends you could spawn a separate process with some small nodejs application to automatically upload that video into a new opencast event
- Creating a new background service for plugNmeet-server, that can listen to STOP_RECORDING events and can automatically upload recordings to opencast.
PS. I can also try to implement this feature myself.

**Comment by jibon57** [comment]
@Lyx52 I think first idea will be easier. In this case, can be include in the config file of recorder a new option e.g `post_processing_scripts` where you can put all your script paths. When [this method](https://github.com/mynaparrot/plugNmeet-recorder/blob/d329bad22c19cb758b3b290f7803148acc2b414c/src/utils/recordingService.ts#L153C5-L153C47) will call during that time we can start a new `spawn` or `fork` of those scripts. I really don't want to add dependencies with recorder package manager. Those scripts should need to be fully separate & managed by it's own with own dependencies.

**Reply by Lyx52** [reply]
Sure probably would be the best if these kinds of projects are kept separate. I guess we can forward some json payload to these scripts for example information about filename, path size recording id etc. 
Is there a way to request room information about inactive/ended rooms? I would probably need some room info like title, creation time.

**Reply by jibon57** [reply]
I've already started working on it. Let me push when it's ready, after that we can discuss more.

**Reply by jibon57** [reply]
@Lyx52 https://github.com/mynaparrot/plugNmeet-recorder/pull/115

**Comment by Lyx52** [comment]
I have created a script for uploading plugnmeet recordings directly into opencast 
https://github.com/Lyx52/plugNmeet-opencast-ingester
